### PR TITLE
Add types declaration file

### LINKED ISSUE #3: TypeScript type declaration file
If you want I can write a TypeScript type declaration file for this package so that TS developers can utilise the type-checking feature of the language as they use this package. Let me know if you are interested.

### PR DESCRIPTION
**Pull request by geopic**
Resolves #3. Let me know if there are any issues.

### CODE REVIEW

File: src/EtsyClient.d.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+declare module 'node-etsy-client' {
+  type EtsyClientOptions = Partial<{
```
[the remark this answers is not in the document]
Comment (boly38) [reply src/EtsyClient.d.ts:2]: hi @geopic
first thanks for this mr 👍 

As said [here](https://stackoverflow.com/questions/55989643/vscode-eslint-complaining-parsing-error-only-declares-and-type-imports-are-allo) could you try to create an `.eslintignore` file with  `/**/*.d.ts` ?
(or else I will amend asap)

[the remark this answers is not in the document]
Comment (geopic) [reply src/EtsyClient.d.ts:2]: Sure, I'll get that sorted.

**Review by creharmony — APPROVED** [review]
I will merge.

Ijn addition I would like to precise: I were trying to add a typescript dedicated unit test but seems more complex than I though because I've no enough experience with TS (need dedicated subdirectory with `package.json` including `type:module`?). 
So in state I've no way to garantee the non regression of ts compatibility. If you know how to do that, please give me some hint (or PR :))

**Comment by creharmony** [comment]
👍 thanks for your contribution @geopic

**Pull request merged**